### ISSUE TITLE
Change trial's assert method implementation to stop emphasizing the `fail` variations

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5901]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#5901|
|Type|enhancement|
|Created|2012-08-22 18:30:41Z|
|Branch|https://github.com/twisted/twisted/tree/trial-assert-5901|

Everyone is switching to the `assert` spellings, [including us](http://twistedmatrix.com/pipermail/twisted-python/2011-March/023681.html).

<details><summary>Searchable metadata</summary>

```
trac-id__5901 5901
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_trial_assert_5901 branches/trial-assert-5901
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__ 
time__1345660241000000 1345660241000000
changetime__1345721727000000 1345721727000000
version__None None
owner__exarkun exarkun
cc__jml
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [35336]) Branching to 'trial-assert-5901'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

(In [35337]) Rename the assertion methods to start with "assert" and make "fail" be the aliases.

refs [#5901](https://github.com/twisted/twisted/issues/5901)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun removed owner|
|-|-|

[build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/trial-assert-5901)

**Comment by twisted-trac** [comment]
|[<img alt="Julian's avatar" src="https://avatars.githubusercontent.com/u/329822?s=50" width="50" height="50">](https://github.com/Julian)|@Julian set owner to @Julian|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="Julian's avatar" src="https://avatars.githubusercontent.com/u/329822?s=50" width="50" height="50">](https://github.com/Julian)<a name="note_6"></a>|@Julian set owner to @exarkun|
|-|-|

Looks good to me. Only thing worth pointing out that I see is that `assertWarns`'s `def` line will fit within 79 chars now.

Also, `assertNotIsInstance`'s docstring is wrong. You're not changing that, so I guess that deserves another ticket if you don't want to fix it while you're here, or if you'd prefer that anyhow. The reason I mention it is that now that the assertion methods are defined using positive language, perhaps all of the docstrings might be clearer if they used that too so that you don't need any mental gymnastics to read them. That can go in another ticket too I guess if that's something reasonable sounding.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun commented|
|-|-|

(In [35347]) Fix an incorrect negation in assertNotIsInstance

refs [#5901](https://github.com/twisted/twisted/issues/5901)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun set status to `closed`|
|-|-|

(In [35348]) Merge trial-assert-5901

Author: exarkun
Reviewer: Julian
Fixes: [#5901](https://github.com/twisted/twisted/issues/5901)

Change trial's implementation of assertion methods so that they are named
following the "assertXYZ" convention rather than the "failXYZ" convention.
